db/db_test.go

```diff
@@ -227,6 +228,147 @@ func TestPut(t *testing.T) {
 	mustGetVersion(ver3, "test value 2")
 }
 
+func TestCreateVersion(t *testing.T) {
+	d := setectest.NewDB(t, nil)
+	id := d.Superuser
+
+	secretName := "secret1"
+	checkVersion := func(version api.SecretVersion, want []byte) *api.SecretValue {
+		t.Helper()
+		got := d.MustGetVersion(id, secretName, version)
+		if !bytes.Equal(got.Value, want) {
+			t.Fatalf("Get %q version %v: got %q, want %q", secretName, version, string(got.Value), string(want))
+		}
+		return got
+	}
+	checkActiveVersion := func(want []byte) {
+		t.Helper()
+		got := d.MustGet(id, secretName)
+		if !bytes.Equal(got.Value, want) {
+			t.Fatalf("Get active %q: got %q, want %q", secretName, string(got.Value), string(want))
+		}
+	}
+
+	testValue1 := []byte("test value 1")
+	testValue2 := []byte("test value 2")
+	testValue3 := []byte("test value 3")
+
+	resetDB := func() {
+		d = setectest.NewDB(t, nil)
+	}
+
+	// One use for setting explicit versions is doing time-based rotation using something like
+	// UNIX timestamps. This simulates that.
+	year2099 := api.SecretVersion(time.Date(2099, 12, 31, 24, 60, 60, 0, time.UTC).Unix())
+
+	t.Run("create", func(t *testing.T) {
```

Comment:
Nit: The convention we used in the existing tests was MixedCase or MixedCase_withQualifier.